lib/autoload/active_record/associations/preloader/manual_has_many_through.rb

```diff
@@ -0,0 +1,58 @@
+class ActiveRecord::Associations::Preloader
+  class ManualHasManyThrough < HasManyThrough
+    def initialize(klass, owners, reflection, records)
+      super(klass, owners, reflection, nil)
+
+      @records_by_owner = records.each_with_object({}) do |record, h|
+        owner_id = record[source_reflection.foreign_key]
+        records = (h[owner_id] ||= [])
+        records << record
+      end
+    end
+
+    def associated_records_by_owner(preloader)
+      preloader.preload(owners,
```

Comment:
This method is copied from `HasManyThrough` itself. And I just modified one line. It makes sense to comment which line I modified but is that necessary to separate this method or comment each line?